core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -0,0 +1,649 @@
+   */
+  async goToSettingsTab(): Promise<void> {
+    if (this.isViewportAtMobileWidth()) {
+      await this.page.waitForSelector(navBarOpener, {visible: true});
+      await this.clickOn(navBarOpener);
+      await this.clickOn(optionsDropDown);
+      await this.clickOn(mobileSettingsBar);
+      /**
+       * Open all dropdowns.
+       */
+      await this.clickOn(basicSettingsDropDown);
+      await this.clickOn(advanceSettingsDropDown);
+      await this.clickOn(rolesSettingsDropDown);
+      await this.clickOn(voiceArtistSettingsDropDown);
+      await this.clickOn(permissionSettingsDropDown);
+      await this.clickOn(feedbackSettingsDropdown);
+      await this.clickOn(explorationControlsSettingsDropDown);
+    } else {
+      await this.clickOn(settingsTab);
+    }
+  }
+
+  /**
+   * This function helps in updating Title.
+   */
+  async addTitle(Title: string): Promise<void> {
+    await this.page.waitForSelector(`${addTitleBar}:not([disabled])`);
+    await this.type(addTitleBar, Title);
+    showMessage('Title has been updated to ' + Title);
+  }
+
+  /**
+   * This function checks length of title bar at basic settings tab.
+   */
+  async expectTitleToHaveMaxLength(maxLength: number): Promise<void> {
+    const titleInput = await this.page.$('.e2e-test-exploration-title-input');
+    const title = await this.page.evaluate(input => input.value, titleInput);
+    const titleLength = title.length;
+
+    if (titleLength <= maxLength) {
+      showMessage(
+        'Title length is within the' +
+          ` allowed limit of ${maxLength} characters.`
+      );
+    } else {
+      throw new Error(
+        'Title length exceeds the allowed' +
+          ` limit of ${maxLength} characters.`
+      );
+    }
+  }
+
+  /**
+   * This function helps in adding a goal.
+   */
+  async updateGoalTo(goal: string): Promise<void> {
+    await this.clickOn(addGoal);
+    await this.type(addGoal, goal);
+  }
+
+  /**
+   * This function checks if the goal has been set in the exploration.
+   */
+  async expectGoalToEqual(expectedGoal: string): Promise<void> {
+    try {
+      const goalInput = await this.page.$('#explorationObjective');
+      if (!goalInput) {
+        throw new Error('Goal input element not found.');
+      }
+
+      const goal = await this.page.evaluate(input => input.value, goalInput);
+
+      if (goal === expectedGoal) {
+        showMessage('The goal has been set for the exploration.');
+      } else {
+        throw new Error('The goal does not match the expected goal.');
+      }
+    } catch (error) {
+      console.error('Error:', error.message);
+      throw error;
+    }
+  }
+
+  /**
+   * This function helps in selecting a category from dropdown.
+   */
+  async selectACategory(category: string): Promise<void> {
+    await this.clickOn(categoryDropDown);
+    await this.clickOn(category);
+  }
+
+  /**
+   * This function checks if a category has been selected for the exploration.
```

Comment:
This docstring is actually wrong. The function is checking if the category matches the expected category, not if a category has been selected.

Please don't write misleading docstrings. Check your docstrings in this file and verify their correctness.